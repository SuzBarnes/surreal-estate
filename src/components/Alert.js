import React from "react";
import "../styles/alert.css";

const Alert = () => {
    return ("Hi")
}

export default Alert;