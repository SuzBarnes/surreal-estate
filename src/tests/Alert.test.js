import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
    it("logs 'Error!' when there is an error", () => {
        const { getByText, asFragment } = render(<Alert message="Error!" error />);

        expect(getByText(/error/i).textContent).toBe("Error!");
        expect(asFragment()).toMatchSnapshot();
    });

    it("logs 'Success!!!' when the task is successful", () => {
        const { getByText, asFragment } = render(<Alert message="Success!!!" success />);

        expect(getByText(/success/i).textContent).toBe("Success!!!");
        expect(asFragment()).toMatchSnapshot();
    });

    it("does not render Alert component when message is not truthy", () => {
        const { asFragment } = render(<Alert message="" />);
        const alert = asFragment();

        expect(alert).toMatchSnapshot();
    })
});
