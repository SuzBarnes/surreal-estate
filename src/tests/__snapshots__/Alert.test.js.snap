// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render Alert component when message is not truthy 1`] = `<DocumentFragment />`;

exports[`Alert logs 'Error!' when there is an error 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert logs 'Success!!!' when the task is successful 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!!!
  </div>
</DocumentFragment>
`;
